handlers/restart_consul.yml

```diff
@@ -6,12 +6,11 @@
   when: ansible_service_mgr == "systemd"
   listen: 'reload systemd daemon'
 
-- name: Restart consul on Unix
-  service:
-    name: consul
-    state: restarted
-    # Needed to force SysV service manager on Docker for Molecule tests
-    use: "{{ ansible_service_mgr }}"
+- name: Rolling restart consul on Unix
+  # Consul leave will perform a clean stop while adding delay between restarts (combined
+  # with the throttle option) so an existing cluster nodes will stay up & running.
+  throttle: 1
+  shell: "{{ consul_binary }} leave {% if consul_acl_enable %} -token {{ consul_acl_master_token }} {% endif %} -http-addr {{ consul_addresses_http }}:{{ consul_ports_http }} && systemctl restart consul.service"
```

Comment:
Unfortunately this PR won't work as presently constructed. Dropping the `service` task in favor of a hardcoded `systemctl restart` command means that this will break on non-systemd Linux hosts. Also, although CI isn't working at the moment (I need to look into fixing that when I have some time), [ansible-lint would complain about the `shell` module](https://ansible.readthedocs.io/projects/lint/rules/command-instead-of-shell/). I guess a suppression would be necessary, since the `&&` requires `shell` and can't be cleanly replaced by `command`.